pkg/api/peer_req_resp.go

```diff
@@ -44,3 +48,25 @@ Example:
 Note - Cannot use query parameters if peerid is also supplied.
 */
 type PeerListResp []PeerGetResp
+
```

Comment:
As both these are in the same package, you can take the common code out to a function:

```go
func mapSize(metadata map[string]string) int {
    size := 0
    for key, value := range p.Metadata {
        ...
    }
    return size
}

func (p *PeerAddReq) MetadataSize() int {
    return mapSize(p.Metadata)
}

func (p *PeerEditReq) MetadataSize() int {
    return mapSize(p.Metadata)
}
```

Reply:
Do the same for volume structs too as it's in the same `api` package. You can put `mapSize` in `pkg/api/utils.go`